source/fundamentals/query-data.txt

```diff
@@ -0,0 +1,194 @@
+.. _entity-framework-query-data:
+
+==========
+Query Data
+==========
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: EF, EF Core, code example, read
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+{+framework-core+} allows you to work with data in your application without
+explicitly running database commands. To query your data, use the
+Langage-Integrated Query (LINQ) syntax. LINQ allows you to write strongly typed
+queries using C#-specific keywords and operators. When you run the application,
+the {+provider-long+} automatically translates the LINQ queries and runs them on the
+database using the MongoDB Query API.
+
+In this guide you can see examples of common query operations on
+an application configured to use the {+provider-long+}.
+
+.. tip::
+
+   To learn how to configure an application to use the {+provider-short+}, see
+   :ref:`entity-framework-configure`.
+
+Find Entities
+-------------
+
+Find a single entity by using the ``FirstOrDefault()`` method, or find
+multiple entities by using the ``Where()`` and ``ToList()`` methods.
+
+Find a Single Entity
+~~~~~~~~~~~~~~~~~~~~
+
+The ``FirstOrDefault()`` method returns the first entity it finds in your collection that
+matches the search criteria, and returns ``null`` if no matching entities
+are found.
+
+The following code uses the ``FirstOrDefault()`` method to find a planet with
+the ``name`` field of "Mercury" from a ``DBSet`` called ``Planets`` and prints
+the planet name to the console:
+
+.. literalinclude:: /includes/code-examples/quick-reference.cs
+   :start-after: // start-find-one
+   :end-before: // end-find-one
+   :language: csharp
+   :dedent:
+
+To learn more about the ``FirstOrDefault()`` method, see the `.NET API documentation
+<https://learn.microsoft.com/en-us/dotnet/api/system.linq.queryable.firstordefault?view=net-8.0>`__.
+
+Find Multiple Entities
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``Where()`` method to retrieve multiple entities from your
+collections. ``Where()`` returns all entities that match the search
+criteria.
+
+The following code uses the ``Where()`` method to find all planets that have the
+``hasRings`` field set to ``true`` and prints the planet names to the console.
+
+.. literalinclude:: /includes/code-examples/quick-reference.cs
+   :start-after: // start-find-many
+   :end-before: // end-find-many
+   :language: csharp
+   :dedent:
+
+To learn more about the ``Where()`` method, see the `.NET API documentation
+<https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.query.queryablemethods.where>`__.
+
+Find All Entities
+~~~~~~~~~~~~~~~~~
+
+You can get all entities from your database by using the ``ToList()`` method.
+The ``ToList()`` method will return all entities in the specified collection as
+a ``List``.
+
+The following code uses the ``ToList()`` method to return all entities in a
```

Comment:
I don't think we need to talk about ToList explicitly - there isn't anything in Linq to deal with it it just causes the query to execute by virtue of enumerating the collection. 

I'd recommend we say it is caused by evaluating the collection and remove the ToList from the code sample and MSDN reference as well.

Reply:
Removed ToList from all code examples (and in the quick reference) and removed the Find All Entities section from this page since, looking over it again, I don't think it really has much value here since it's not really a query